### ISSUE TITLE
Question regarding multiple requests to an ensemble

### ISSUE THREAD

**Issue by goncalo-maia-98** [triton-inference-server/server#6170]
**Description**
I was looking for clarification about multiple requests to a model ensemble, in particular for the execution order and parallel operations. If a ensemble consists of multiple models and receives multiple requests will the server wait for the full ensemble to finish before scheduling the second request or will the second request use the first model of the ensemble as soon as this terminates with the first request. 

As an example the model from the [documentation on ensembles](https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/user_guide/architecture.html#ensemble-models),  where we have a pre-processing model followed by a classification and a segmentation models in parallel. <br>
If two inference requests arrive for this ensemble will the server wait for the ensemble to conclude before processing the second request, or, will the second request use the pre-processing model as soon as this finishes with the first but the segmentation and classification are still processing information from the first request. 

Hope my explanation of the question was understandable, I will be happy to clarify if necessary. 
Thank you.

**Comment by nv-kmcgill53** [comment]
Hi @goncalo-maia-98, assuming in your example the two requests are not batched together, the answer depends on the number of instances of each model which makes up the ensemble. If you have an ensemble of `A-> B-> C` all with one instance each, the first request gets scheduled on `A` and the second request must wait until `A` is finished before being scheduled. If, however, `A` has two instances then the first request will be scheduled on instance 1 of `A` and the second request will be scheduled on instance 2 of `A`. The same behavior will happen for the downstream models, `B` and `C`, as well (assuming no batching). 

in short, no, the entire ensemble doesn't block requests until complete, only the model instances which make up the ensemble will block if there are requests processing. Once an instance is free it will check the scheduler to get the next request regardless of the state of the ensemble.
